lib/Fmt_ast.ml

```diff
@@ -341,37 +341,37 @@ let fmt_direction_flag = function
   | Upto -> fmt "@ to "
   | Downto -> fmt "@ downto "
 
+let fmt_private c loc = hvbox 0 @@ Cmts.fmt c loc @@ str "private"
+
+let fmt_virtual c loc = hvbox 0 @@ Cmts.fmt c loc @@ str "virtual"
+
+let fmt_mutable c loc = hvbox 0 @@ Cmts.fmt c loc @@ str "mutable"
+
 let fmt_private_flag c = function
-  | Private loc -> fmt " " $ Cmts.fmt c loc @@ str "private"
+  | Private loc -> fmt "@ " $ fmt_private c loc
   | Public -> noop
 
 let fmt_virtual_flag c = function
-  | Virtual loc -> fmt " " $ Cmts.fmt c loc @@ str "virtual"
+  | Virtual loc -> fmt "@ " $ fmt_virtual c loc
   | Concrete -> noop
 
 let fmt_mutable_flag c = function
-  | Mutable loc -> Cmts.fmt c loc @@ str "mutable" $ fmt " "
+  | Mutable loc -> fmt_mutable c loc $ fmt "@ "
   | Immutable -> noop
 
+let opt_flag ~f c x = opt x (fun x -> fmt "@ " $ f c x)
+
 let fmt_mutable_virtual_flag c = function
   | {mv_mut= Some m; mv_virt= Some v} when Location.compare_start v m < 1 ->
-      fmt " "
-      $ Cmts.fmt c v @@ str "virtual"
-      $ fmt " "
-      $ Cmts.fmt c m @@ str "mutable"
+      opt_flag c (Some v) ~f:fmt_virtual $ opt_flag c (Some m) ~f:fmt_mutable
```

Comment:
The `fmt "@ "` of `opt_flag` could be a `?(epi=fmt "@ ")` argument of the `fmt_private/virtual/mutable` functions instead. This removes a falsely general function that is only used for this. `fmt_mutable_flag` can be made more consistent also.